platform/pulpcore/download/context.py

```diff
@@ -0,0 +1,223 @@
+import sys
+
+from gettext import gettext as _
+from logging import getLogger
+from threading import RLock
+from datetime import datetime, timedelta
+
+log = getLogger(__name__)
+
+
+class NotCached(Exception):
```

Comment:
This could use ``LookupError`` as a base. It's more specific, and I think well-suited.